packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -775,6 +776,9 @@ export async function downloadSnapshot(
 	};
 }
 
+const mockableDownloadSnapshot = mockify(downloadSnapshot);
+export { mockableDownloadSnapshot as downloadSnapshot };
```

Comment:
this pattern breaks IntelliSense for the function, and also makes it easy for any code using the function in this package to call either one (and isn't super clear which one should be called.

I think it would be better to avoid having two separate functions in scope here: if you want all callers to use this version, I think a decorator would be a better way to wrap it as that would keep the non-wrapped version out of scope and allows the IntelliSense/docs to function properly. It would also be more concise.

Reply:
Well, as [I mentioned in this comment](https://github.com/microsoft/FluidFramework/pull/23728/files/15d8a18ccde7ff9765e36d36f9f04a0f0b8749e6#r1938120505), I can just inline it instead, so there is no rename+export. I like that better anyway, it just ruined the diff for this PR.

When you say decorator do you mean an actual TS decorator? I was under the impression that you can only write those for class members, not free functions.

Reply:
Apparently, you are right, they don't work on free functions.


If you don't inline them, I think you should nave the version you won't want people using something like fooInner instead of naming the one you do want people to use mockableFoo. Also leave the doc comments on the actually exported one (and maybe link or inherit doc the inner one to it).

I do think inlining it would be better, though refactoring things do we don't need mocking in the first place would be best: adding testing only hooks to production code is always a bit sketchy as it implies the code itself isn't flexible enough to do what tests need which is generally a bad sign.

Reply:
You're right that renaming the inner helper would be better than renaming the outer exported function (because otherwise, code calling the function from _within its module of declaration_ will accidentally use the wrong one). However, I just went ahead and in-lined the calls.

I also took a look at refactoring so that e.g. `fetch` is injected rather than mocked. It's of course possible, but not trivial, and is beyond the scope of this PR IMO.